src/onelogin/saml2/idp_metadata_parser.py

```diff
@@ -60,16 +64,26 @@ def parse_remote(url):
         return OneLogin_Saml2_IdPMetadataParser.parse(idp_metadata)
 
     @staticmethod
-    def parse(idp_metadata):
+    def parse(idp_metadata, preferred_sso_binding=OneLogin_Saml2_Constants.BINDING_HTTP_REDIRECT):
         """
-        Parse the Identity Provider metadata and returns a dict with extracted data
-        If there are multiple IDPSSODescriptor it will only parse the first
+        Parse the Identity Provider metadata and return a dict with extracted data.
+
+        If there are multiple IDPSSODescriptor, parse only the first.
+
+        Parse only those SSO endpoints with the same binding as given by
+        the `preferred_sso_binding` parameter.
+
+        If there are multiple SSO endpoints, extract only the first.
```

Comment:
Your current code is not exracting the first if there are multiple SSO endponts.

I think that we better extract all the SingleSignOnService, and later iterate and see if there is any with the expected binding, if not get the first.

Also, what happen with SLO endpoints?  I thing we should also add a preferred_slo_binding